### ISSUE TITLE
loadImage.js Cross Origin 

### ISSUE THREAD

**Issue by pierotofy** [AnalyticalGraphicsInc/cesium#3638]
The current code does not allow passing of credentials to a map server hosted on a separate domain via CORS.

In Source/Core/loadImage.js:86

``` javascript
if (crossOrigin) {
    image.crossOrigin = ''; // defaults to "anonymous", no credentials are sent
}
```

https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_settings_attributes
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img (crossorigin)

Should there be a way to choose between "anonymous" and "use-credentials"?

This probably affects other areas that rely on image loading via Image object creation.

**Comment by mramato** [comment]
@pierotofy You are correct, but we need to figure out a clean way to implement it (which is the main reason it's not done yet).  The two ideas would be global per-origin settings on the `loadImage` function itself or threading through options from the top all the way down into `loadImage`.  I've wavered back and forth between the two but would have to try them to know which is ultimately better.

**Comment by pierotofy** [comment]
I think an approach similar to how BingMapsApi keys are set could work well:

```javascript
BingMapsApi.defaultKey = "My key";

loadImage.crossOriginAttribute = loadImage.CROSS_ORIGIN_ANONYMOUS;
```

Or maybe add a method:

```javascript
loadImage.corsSendCredentials(true);
```

I wouldn't see a need for cascading the options from the top, unless there's a use case where credentials should be sent only on certain requests and not in others.

**Comment by mramato** [comment]
> unless there's a use case where credentials should be sent only on certain requests and not in others.

I believe that at the very least, it would need to be per-origin, so instead of a single value it's an object whose keys are the origin and value is the value assigned to crossOriginAttribute.

CC @shunter any thoughts here?

**Comment by mramato** [comment]
Yes, same problem (though the global options approach may not be as appropriate here but give it a try and see).  I would not open a separate issue.

**Comment by mramato** [comment]
Is this addressed by #4298?  If not, it may be easy to add using the same registry introduced there.

**Comment by pjcozzi** [comment]
> Is this addressed by #4298? If not, it may be easy to add using the same registry introduced there.

Closing. @pierotofy please reopen if there is still an issue.

**Comment by pierotofy** [comment]
Yes the addition of `TrustedServer` has solved this. Thank you!!
